tests/integration/test_sqs.py

```diff
@@ -1194,6 +1194,390 @@ def receive_and_check_order():
+        response = aws_client.sqs.receive_message(
+            QueueUrl=queue_url,
+            MaxNumberOfMessages=10,
+        )
+        ordering = [message["Body"] for message in response["Messages"]]
+
+        # these are in principle all valid orderings
+        assert ordering in (
+            ["g2-m1", "g2-m2", "g1-m3", "g1-m4", "g3-m1"],
+            ["g2-m1", "g2-m2", "g3-m1", "g1-m3", "g1-m4"],
+            ["g3-m1", "g2-m1", "g2-m2", "g1-m3", "g1-m4"],
+            ["g1-m3", "g1-m4", "g3-m1", "g2-m1", "g2-m2"],
+        )
+
+    @pytest.mark.aws_validated
+    def test_fifo_message_group_visibility_after_partial_delete(self, sqs_create_queue, aws_client):
+        # this is almost the same test case as the one above, only that it doesn't fully delete the messages
+        # it received so the message group remains in-flight.
+        queue_url = sqs_create_queue(
+            QueueName=f"queue-{short_uid()}.fifo",
+            Attributes={
+                "FifoQueue": "true",
+                "ContentBasedDeduplication": "true",
+            },
+        )
+
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g1-m1", MessageGroupId="group-1"
+        )
+        # we interleave a message from group 2
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g2-m1", MessageGroupId="group-2"
+        )
+        # continue with group 1
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g1-m2", MessageGroupId="group-1"
+        )
+        # we interleave another a message from group 2
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g2-m2", MessageGroupId="group-2"
+        )
+        # continue with group 1
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g1-m2", MessageGroupId="group-1"
+        )
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g1-m3", MessageGroupId="group-1"
+        )
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g1-m4", MessageGroupId="group-1"
+        )
+        # add group 3
+        aws_client.sqs.send_message(
+            QueueUrl=queue_url, MessageBody="g3-m1", MessageGroupId="group-3"
+        )
+
+        # we receive two messages from group 1 (note that group 2 is not in there)
+        response = aws_client.sqs.receive_message(QueueUrl=queue_url, MaxNumberOfMessages=2)
+        assert response["Messages"][0]["Body"] == "g1-m1"
+        assert response["Messages"][1]["Body"] == "g1-m2"
+
+        # we delete only one in-flight messages we received from group 1
+        aws_client.sqs.delete_message(
+            QueueUrl=queue_url, ReceiptHandle=response["Messages"][0]["ReceiptHandle"]
+        )
+
+        # draining the queue should now not the message from group 1, since it's not yet visible
```

Comment:
nit: missing word (I guess "not include the message")

Reply:
it _should_ :-) i rephrased it a bit